src/protagonist/Engine/Ingest/Image/ImageServer/Clients/CantaloupeThumbsClient.cs

```diff
@@ -0,0 +1,86 @@
+﻿using DLCS.Core.Exceptions;
+using DLCS.Core.FileSystem;
+using DLCS.Core.Strings;
+using DLCS.Core.Types;
+using DLCS.Model.Templates;
+using Engine.Ingest.Image.ImageServer.Manipulation;
+using Engine.Settings;
+using Microsoft.Extensions.Options;
+
+namespace Engine.Ingest.Image.ImageServer.Clients;
+
+public class CantaloupeThumbsClient : ICantaloupeThumbsClient
+{
+    private readonly HttpClient thumbsClient;
+    private readonly EngineSettings engineSettings;
+    private readonly IFileSystem fileSystem;
+    private readonly IImageManipulator imageManipulator;
+        
+    public CantaloupeThumbsClient(
+        HttpClient thumbsClient,
+        IFileSystem fileSystem,
+        IImageManipulator imageManipulator,
+        IOptionsMonitor<EngineSettings> engineOptionsMonitor)
+    {
+        this.thumbsClient = thumbsClient;
+        engineSettings = engineOptionsMonitor.CurrentValue;
+        this.fileSystem = fileSystem;
+        this.imageManipulator = imageManipulator;
+    }
+
+    public async Task<List<ImageOnDisk>> CallCantaloupe(IngestionContext context,
+        List<string> thumbSizes, 
+        CancellationToken cancellationToken = default)
+    {
+        var thumbsResponse = new List<ImageOnDisk>();
+        
+        var convertedS3Location = context.ImageLocation.S3.Replace("/", engineSettings.ImageIngest!.ThumbsProcessorSeparator);
+
+        foreach (var size in thumbSizes!)
+        {
+            using var response =
+                await thumbsClient.GetAsync(
+                    $"iiif/3/{convertedS3Location}/full/{size}/0/default.jpg", cancellationToken);
+
+            if (response.IsSuccessStatusCode)
+            {
+                await using var responseStream = await response.Content.ReadAsStreamAsync();
+                var assetDirectoryLocation = Path.GetDirectoryName(context.AssetFromOrigin.Location);
+
+                var localThumbsPath =
+                    $"{assetDirectoryLocation}{Path.DirectorySeparatorChar}output{Path.DirectorySeparatorChar}thumbs{Path.DirectorySeparatorChar}{size}";
+                
+                await fileSystem.CreateFileFromStream(localThumbsPath, responseStream, cancellationToken);
+                
+                var image = await imageManipulator.LoadAsync(localThumbsPath, cancellationToken);
+
+                thumbsResponse.Add(new ImageOnDisk()
+                {
+                    Path = localThumbsPath,
+                    Width = image.Width,
+                    Height = image.Height
+                });
+            }
+            else
+            {
+                throw new HttpException(response.StatusCode, "failed to retrieve data from the thumbs processor");
```

Comment:
Where does this message end up? If looking in logs would I see what request failed from messages around it? If not it might be worth including in the error.

Reply:
it would bubble up to `ProcessImage` in `ImageServerClient`